QuizApp/QuizApp/UI/Components/ProgressView.swift

```diff
@@ -0,0 +1,54 @@
+import UIKit
+
+class ProgressView: UIView {
+
+    private var stackView: UIStackView!
+
+    override init(frame: CGRect) {
+        super.init(frame: .zero)
+
+        createViews()
+        styleViews()
+        defineLayoutForViews()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func set(colors: [UIColor]) {
+        stackView
+            .arrangedSubviews
+            .forEach { $0.removeFromSuperview() }
+
+        for color in colors {
+            let view = UIView()
+            view.backgroundColor = color
+            view.layer.cornerRadius = 2
+            stackView.addArrangedSubview(view)
+        }
+    }
+
+}
+
+extension ProgressView: ConstructViewsProtocol {
+
+    func createViews() {
+        stackView = UIStackView()
+        addSubview(stackView)
+    }
+
+    func styleViews() {
+        stackView.axis = .horizontal
+        stackView.spacing = 8
+        stackView.distribution = .fillEqually
+    }
+
+    func defineLayoutForViews() {
+        stackView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+            $0.height.equalTo(5)
```

Comment:
It can be misleading to add top & bottom constraints and height. You probably want that this whole view has a height of 5. So, it would be more clear to write:
```
func defineLayoutForViews() {
    snp.makeConstraints {
        $0.height.equalTo(5)
    }

    stackView.snp.makeConstraints {
        $0.edges.equalToSuperview()
    }
}
```